api/src/test/cards.test.ts

```diff
@@ -327,5 +385,160 @@ describe('Testing Cards', () => {
+      // const result = await request(app.getServer()).get(`${cardsRoute.path}/${cardId}?include[]=price-history&include[]=card-rating`);
+      // const result = await request(app.getServer()).get(`${cardsRoute.path}/${cardId}`)
+      //   .query({
+      //     include: [`price-history`, `card-rating`]
+      //   });
+      const result = await request(app.getServer()).get(`${cardsRoute.path}/${cardId}?include=price-history;card-rating`);
+
+      // Expect good status code and cardId to match the route
+      expect(result.status).toEqual(200);
+      // Note had to use toString because of objectId type.
+      expect(result.body.data._id.toString()).toEqual(cardId.toString());
+
+      // Check pokemon card fields 
+      expect(result.body.data.name).toEqual(createdCards[1].name);
+      expect(result.body.data.description).toEqual(createdCards[1].description);
+      expect(result.body.data.salePrice).toEqual(createdCards[1].salePrice);
+      expect(result.body.data.marketPrice).toEqual(createdCards[1].marketPrice);
+      // expect(result.body.data.rating).toEqual(createdCards[1].rating);
+      expect(result.body.data.image).toEqual(createdCards[1].image);
+
+      // Check second price history entry
+      const dateAndTimePriceHistory1 : string[] = result.body.data.priceHistoryEntries[1].date.split("T"); 
+      const dateOfPriceHistory1 : string = dateAndTimePriceHistory1[0]; 
+      const [year1, month1, day1] : string[] = dateOfPriceHistory1.split('-');
+      const responseDateObject1 = new Date(+year1, +month1 - 1, +day1);
+
+      expect(responseDateObject1.toString()).toBe(priceHistoryData1.date.toString());
+      // expect(result.body.data.priceHistoryEntries[0]._id.toString()).toEqual(createdPriceHistory1._id.toString());
+      expect(result.body.data.priceHistoryEntries[1].pokemonCardId.toString()).toEqual(priceHistoryData1.pokemonCardId.toString());
+      expect(result.body.data.priceHistoryEntries[1].price).toEqual(priceHistoryData1.price);
+      expect(result.body.data.priceHistoryEntries[1].quantity).toEqual(priceHistoryData1.quantity);       
+
+      // Check first card rating entry
+      const dateAndTimeCardRating0 : string[] = result.body.data.cardRatingEntries[0].date.split("T"); 
+      const dateOfCardRating0 : string = dateAndTimeCardRating0[0]; 
+      const [year0, month0, day0] : string[] = dateOfCardRating0.split('-');
+      const responseDateObject0 = new Date(+year0, +month0 - 1, +day0);
+
+      expect(responseDateObject0.toString()).toBe(cardRatingData0.date.toString());
+      // expect(result.body.data.priceHistoryEntries[0]._id.toString()).toEqual(createdPriceHistory1._id.toString());
+      expect(result.body.data.cardRatingEntries[0].pokemonCardId.toString()).toEqual(cardRatingData0.pokemonCardId.toString());
+      expect(result.body.data.cardRatingEntries[0].rating).toEqual(cardRatingData0.rating);
+
+
+    })
+
   });
 });
```

Comment:
Really good, but there is a lot of commented out code that can be deleted to clean up the file.

Reply:
Yeah true. I just went through and deleted all the dirty unneeded commented code